internal/httpserver/server.go

```diff
@@ -0,0 +1,264 @@
+type ServerOptions struct {
+	// HTTPServerAddress is the address where the server will listen for
+	// incoming requests.
+	// It should be in the format "host:port" (e.g., ":8080").
+	// Required.
+	HTTPServerAddress string
+
+	// CertFile is the path to the TLS certificate file. If not provided, the
+	// server will run without TLS.
+	// Optional.
+	CertFile string
+
+	// KeyFile is the path to the TLS key file. If not provided, the server
+	// will run without TLS.
+	// Optional.
+	KeyFile string
+
+	// VerifyTimeout is the duration to wait for a verification request to
+	// complete before timing out. Default is 5 seconds if not specified.
+	// Optional.
+	VerifyTimeout time.Duration
+}
+
+// StartServer initializes and starts the Ratify server with provided options
+// and configuration file path.
+func StartServer(opts *ServerOptions, configPath string) error {
+	executorOpts, err := config.Load(configPath)
+	if err != nil {
+		logrus.Errorf("Failed to load executor options: %v", err)
+		return err
+	}
+
+	server, err := newServer(opts, executorOpts)
+	if err != nil {
+		logrus.Errorf("Failed to create server: %v", err)
+		return err
+	}
+
+	logrus.Infof("Starting server at port: %s", opts.HTTPServerAddress)
+	return server.Run()
+}
+
+func newServer(serverOpts *ServerOptions, executorOpts *executor.Options) (*server, error) {
+	e, err := executor.NewExecutor(executorOpts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create executor: %w", err)
+	}
+	server := &server{
+		router:               mux.NewRouter(),
+		address:              serverOpts.HTTPServerAddress,
+		certFile:             serverOpts.CertFile,
+		keyFile:              serverOpts.KeyFile,
+		verifyRequestTimeout: serverOpts.VerifyTimeout,
+		executor:             e,
+	}
+	if server.verifyRequestTimeout == 0 {
+		server.verifyRequestTimeout = defaultVerifyTimeout
+	}
+	if err := server.registerHandlers(); err != nil {
+		return nil, fmt.Errorf("failed to register handlers: %w", err)
+	}
+	return server, nil
+}
+
+func (s *server) registerHandlers() error {
+	if err := s.registerVerifyHandler(); err != nil {
+		return err
+	}
+	if err := s.registerMutateHandler(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// TODO: implement mutate handler.
+func (s *server) registerMutateHandler() error {
+	return nil
+}
+
+func (s *server) registerVerifyHandler() error {
+	verifyURL, err := url.JoinPath(serverRootURL, verifyPath)
+	if err != nil {
+		return err
+	}
+	s.router.Methods(http.MethodPost).Path(verifyURL).Handler(middlewareWithTimeout(s.verifyHandler(), s.verifyRequestTimeout))
+	return nil
+}
+
+func (s *server) verifyHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		_ = s.verify(r.Context(), w, r)
+	}
+}
+
+// verify handles the verification request from Gatekeeper.
+func (s *server) verify(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
```

Comment:
Is it better to keep the function for verify logic only and move the http I/O in `verifyHandler`?
```suggestion
func (s *server) verify(ctx context.Context, requestBody []byte) (externaldata.ProviderResponse， error) {
```